pkg/test/framework/components/echo/echotest/run.go

```diff
@@ -61,6 +64,28 @@ func (t *T) Run(testFn oneToOneTest) {
 	})
 }
 
+// RunFromClusters will generate and run one subtest to send traffic to
+// destination instance. This is for ingress gateway testing when source instance
+// is sending requests to destination through the ingress gateway.
+func (t *T) RunFromClusters(testFn oneClusterOneTest) {
+	t.fromEachClusterOnly(t.rootCtx, func(ctx framework.TestContext, c cluster.Cluster) {
+		t.toEachDeployment(ctx, func(ctx framework.TestContext, dstInstances echo.Instances) {
+			t.setupDst(ctx, dstInstances)
+			testFn(ctx, c, dstInstances)
+		})
+	})
+}
+
+// fromEachClusterOnly runs test from each cluster without requiring source deployment.
+func (t *T) fromEachClusterOnly(ctx framework.TestContext, testFn perClusterTest) {
```

Comment:
nit: this could be called `fromEachCluster` and the existing method could be `fromEachWorkloadCluster` or something.

Reply:
Done.